pkg/controller/windowsmachineconfig/nodeconfig/network.go

```diff
@@ -0,0 +1,130 @@
+package nodeconfig
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// cniConf contains the structure of the CNI template
+type cniConf struct {
+	CNIVersion   string       `json:"cniVersion"`
+	Name         string       `json:"name"`
+	Type         string       `json:"type"`
+	Capabilities capabilities `json:"capabilities"`
+	IPAM         ipam         `json:"ipam"`
+	Policies     policies     `json:"policies"`
+}
+
+// nested structs required for creating
+// cniConf struct
+type capabilities struct {
+	DNS bool `json:"dns"`
+}
+
+type ipam struct {
+	Type   string `json:"type"`
+	Subnet string `json:"subnet"`
+}
+
+type policies []struct {
+	Name  string `json:"name"`
+	Value value  `json:"value"`
+}
+
+type value struct {
+	Type              string   `json:"Type"`
+	ExceptionList     []string `json:"ExceptionList,omitempty"`
+	DestinationPrefix string   `json:"DestinationPrefix,omitempty"`
+	NeedEncap         bool     `json:"NeedEncap"`
+}
+
+// network struct contains the node network information
+type network struct {
+	// hostSubnet holds the node host subnet value
+	hostSubnet string
+}
+
+// newNetwork returns a pointer to the network struct
+func newNetwork() *network {
+	return &network{}
+}
+
+// setHostSubnet sets the value for hostSubnet field in the
+// network struct
+func (nw *network) setHostSubnet(hostSubnet string) error {
+	if hostSubnet == "" || validateCIDR(hostSubnet) != nil {
+		return errors.Errorf("error receiving valid value for node hostSubnet")
+	}
+	nw.hostSubnet = hostSubnet
+	return nil
+}
+
+// validateCIDR uses the parseCIDR from network package
+// to validate the format of the CIDR received
+func validateCIDR(serviceCIDR string) error {
```

Comment:
won't the argument be a generic CIDR or is this validation specific to `serviceCIDR`?